### PR TITLE
chore: fix browser back button not working with Usage Pulse

### PR DESCRIPTION
**Pull request by ankitsrivas14**
Fix the browser back button not working with Usage Pulse because of buggy `history.listen`
>
>
#### PR fixes following issue(s)
Fixes # https://github.com/appsmithorg/appsmith/issues/24892
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by dipyamanbiswas07** [comment]
@RakshaKShetty  could you help verify if all the usagepulse use cases work in this dp?

File: app/client/src/usagePulse/index.ts [hunk 1]
```diff
@@ -93,14 +92,14 @@ class UsagePulse {
    * a callback and unlisten when the user goes to a trackable URL
    */
   static async watchForTrackableUrl(callback: () => void) {
-    UsagePulse.unlistenRouteChange = history.listen(async () => {
+    UsagePulse.unlistenRouteChange = async () => {
```
Comment (sbalaji1192) [inline app/client/src/usagePulse/index.ts:95]: this should be called `routeChangeListener`

**Review by sbalaji1192 — COMMENTED** [review]
Have we found out what was the issue with history listen?

**Review by sbalaji1192 — COMMENTED** [review]
Have we found out what was the issue with history listen?

**Comment by ankitsrivas14** [comment]
> Have we found out what was the issue with history listen?

Not yet.

**Comment by sbalaji1192** [comment]
Have we QA'ed this change?

**Comment by ankitsrivas14** [comment]
Not going forward with this approach as `Popstate` would not work for this use case since the active history is not present on all routes.

This PR would fix the issue https://github.com/appsmithorg/appsmith/pull/25671

**Pull request closed without merging**